hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1845,12 +1851,21 @@ public void move(final byte[] encodedRegionName,
       // closed
       serverManager.sendRegionWarmup(rp.getDestination(), hri);
 
+      // Here wait until all the meta regions are not in transition.
+      if (!hri.isMetaRegion() && assignmentManager.getRegionStates().isMetaRegionInTransition()) {
+        Thread.sleep(timeoutWaitMetaRegionAssignment);
+        if (assignmentManager.getRegionStates().isMetaRegionInTransition()) {
+          LOG.error("This is fail-fast of the region move because "
+            + "hbase:meta region is in transition. Failed region move info: " + rp);
+          return;
```

Comment:
I think we can throw HBaseIOException here since this is fail-fast scenario?

Reply:
Right, exception could be detected by the client. Have changed to throw exception in the latest commit.